src/d/actor/d_a_obj_rforce.cpp

```diff
@@ -4,68 +4,128 @@
-void daObjRforce::Act_c::solidHeapCB(fopAc_ac_c*) {
-    /* Nonmatching */
+BOOL daObjRforce::Act_c::solidHeapCB(fopAc_ac_c* this_i) {
+    return ((Act_c*)this_i)->create_heap() & 0xFF;
 }
 
 /* 0000009C-00000220       .text create_heap__Q211daObjRforce5Act_cFv */
-void daObjRforce::Act_c::create_heap() {
-    /* Nonmatching */
+BOOL daObjRforce::Act_c::create_heap() {
+
+    J3DModelData* mdl_data = static_cast<J3DModelData*>(dComIfG_getObjectRes(M_arcname, VDORA_INDEX_BDL_VDORA));
+    JUT_ASSERT(0x57, mdl_data != NULL);
+    mpModel = mDoExt_J3DModel__create(mdl_data, 0, 0x11000002);
+    
+    set_mtx();
+
+    J3DModelData* bgw_data = static_cast<J3DModelData*>(dComIfG_getObjectRes(M_arcname, 0x07));
+    JUT_ASSERT(0x64, bgw_data != NULL);
+    if(bgw_data != NULL){
+        mpBgw = new dBgW();
+        if(mpBgw != NULL){
+            if(mpBgw->Set((cBgD_t*)bgw_data, cBgW::MOVE_BG_e, &mtx) == true) return FALSE;
+        }
+    }
+
+    BOOL ret = FALSE;
+    if(mdl_data != NULL && mpModel != NULL && bgw_data != NULL && mpBgw != NULL){
+        ret = TRUE;
+    }
+    return ret;
 }
 
 /* 00000220-000002F8       .text _create__Q211daObjRforce5Act_cFv */
 s32 daObjRforce::Act_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, Act_c);
+
+    s32 ret = dComIfG_resLoad(&mPhs, M_arcname);
+    
+    if(ret == cPhs_COMPLEATE_e){
+        if(!fopAcM_entrySolidHeap(this, solidHeapCB, 0) == FALSE){
+            fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+            dComIfG_Bgsp()->Regist(mpBgw, this);
+            mpBgw->SetCrrFunc(dBgS_MoveBGProc_Typical);
+        }
+        else {
+            ret = cPhs_ERROR_e;
+        }
+    }
+
+    return ret;
 }
 
 /* 000002F8-00000384       .text _delete__Q211daObjRforce5Act_cFv */
 BOOL daObjRforce::Act_c::_delete() {
-    /* Nonmatching */
+    if(heap != NULL && mpBgw != NULL && mpBgw->ChkUsed()){
+        dComIfG_Bgsp()->Release(mpBgw);
+    }
+    dComIfG_resDelete(&mPhs, M_arcname);
+    return TRUE;
 }
 
 /* 00000384-00000430       .text set_mtx__Q211daObjRforce5Act_cFv */
 void daObjRforce::Act_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    MTXCopy(mDoMtx_stack_c::get(), mtx);
+    mpModel->calc();
 }
 
 /* 00000430-00000468       .text _execute__Q211daObjRforce5Act_cFv */
 BOOL daObjRforce::Act_c::_execute() {
-    /* Nonmatching */
+    set_mtx();
+    mpBgw->Move();
+    return TRUE;
 }
 
 /* 00000468-00000508       .text _draw__Q211daObjRforce5Act_cFv */
 BOOL daObjRforce::Act_c::_draw() {
-    /* Nonmatching */
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, &tevStr);
+    dComIfGd_setListBG();
+    
+    g_env_light.setLightTevColorType(mpModel, &tevStr);
+    mDoExt_modelUpdateDL(mpModel);
+    dComIfGd_setList();
+
+    return TRUE;
 }
 
+daObjRforce::Act_c::~Act_c(){}
```

Comment:
You can usually put empty destructors in the header instead of the cpp (i.e. `virtual ~Act_c() {}` in the class body)